Source/cursor.cpp

```diff
@@ -147,26 +145,27 @@ void FreeCursor()
 	ClearCursor();
 }
 
-const OwnedCelSprite &GetInvItemSprite(int i)
+const OwnedCelSprite &GetInvItemSprite(int cursId)
 {
-	return i < InvItems1Size ? *pCursCels : *pCursCels2;
+	return cursId <= InvItems1Size ? *pCursCels : *pCursCels2;
 }
 
-int GetInvItemFrame(int i)
+int GetInvItemFrame(int cursId)
 {
-	return i < InvItems1Size ? i : i - (InvItems1Size - 1);
+	return cursId <= InvItems1Size ? cursId - 1 : cursId - InvItems1Size - 1;
 }
 
 Size GetInvItemSize(int cursId)
 {
-	if (cursId >= InvItems1Size)
-		return { InvItemWidth2[cursId - (InvItems1Size - 1)], InvItemHeight2[cursId - (InvItems1Size - 1)] };
-	return { InvItemWidth1[cursId], InvItemHeight1[cursId] };
+	const int i = cursId - 1;
+	if (i >= InvItems1Size)
+		return { InvItemWidth2[i - (InvItems1Size - 1)], InvItemHeight2[i - (InvItems1Size - 1)] };
+	return { InvItemWidth1[i], InvItemHeight1[i] };

```

Comment:
I think the `(InvItem1Size - 1)` value is now off by 1, it was compensating for the 0 size entry that's been removed from the start of that list. Rune of stone is being reported as a 2x2 item in this branch as `ICURS_RUNE_OF_STONE + CURS_FIRST_ITEM - 1 - (179 - 1) == 30`, which I think is meant to be the bovine suit's size.

Reply:
Thanks! Fixed, I think?